system/interceptors/FormBuilderInterceptor.cfc

```diff
@@ -6,4 +6,25 @@ component extends="coldbox.system.Interceptor" {
 	public void function onApplicationStart() {
 		formBuilderService.updateUsesGlobalQuestions();
 	}
+
+	public void function preDeleteObjectData( event, interceptData ) {
+		var objectName = interceptData.objectName ?: "";
+
+		if ( objectName == "formbuilder_formsubmission" ) {
+			var filter = interceptData.filter ?: {};
+			var id     = interceptData.id     ?: ( filter.id ?: "" );
+
+			if ( isArray( id ) ) {
+				id = ArrayFirst( id );
+			}
+
+			if ( !isEmptyString( id ) ) {
+				formBuilderService.deleteSubmissionResponses( submissionId=id );
+			}
+
+			if ( !isEmptyString( filter.form ?: "" ) ) {
```

Comment:
This feels potentially dangerous. *IF* for example, some code ran that did:

```cfc
deleteData( filter={ form=formId, user=userid } );
```

^^ we'd end up deleting entries for the entire form.

Perhaps this?

```cfc
if ( !isEmptyString( filter.form ?: "" ) && StructCount( filter ) == 1 ) {
     // ...?
}
```

Or another approach entirely that should allow most flexibility and coverage:

```cfc
if ( !isEmptyString( id ) ) {
    ...
    return;
}

formBuilderService.deleteFormResponses( responseFilter=filter ); << then use that filter to delete them
```